src/main/java/com/karankumar/bookproject/backend/statistics/PageStatistics.java

```diff
@@ -44,6 +51,53 @@ public Book findBookWithMostPages() {
         return bookWithMostPages;
     }
 
+    /**
+     * @return the Book with the highest number of pages that was read this year
+     */
+    public Optional<Book> findBookReadThisYearWithTheMostPages() {
+        List<Book> booksReadThisYear = booksWithPageCount.stream()
+                                         .filter(PageStatistics::readThisYear)
+                                         .sorted(Comparator.comparing(Book::getNumberOfPages))
+                                         .collect(Collectors.toList());
+
+        return (booksReadThisYear.isEmpty()) ? Optional.empty() :
+                Optional.of(booksReadThisYear.get(booksReadThisYear.size() - 1));
+    }
+
+    private static boolean readThisYear(Book book) {
+        LocalDate dateStarted = book.getDateStartedReading();
+        LocalDate dateFinished = book.getDateFinishedReading();
+        return dateStarted != null && dateIsInCurrentYear(dateStarted) &&
+                dateFinished != null && dateIsInCurrentYear(dateFinished);
+    }
+
+    /**
+     * @return the Book with the lowest number of pages that was read this year
+     */
+    public Book findBookWithLeastPages() {
+        Book bookWithMostPages = null;
+        if (!booksWithPageCount.isEmpty()) {
+            booksWithPageCount.sort(Comparator.comparing(Book::getNumberOfPages));
+            bookWithMostPages = booksWithPageCount.get(0);
+        }
+        return bookWithMostPages;
+    }
+
+    /**
+     * @return the Book in the 'read' shelf with the lowest number of pages this year
+     */
+    public Book findBookWithLeastPagesThisYear() {
+        Book bookWithMostPages = null;
+        booksWithPageCount.removeIf(book -> (book.getDateStartedReading() == null ||
+                !dateIsInCurrentYear(book.getDateStartedReading()))
```

Comment:
How come you're removing elements? This may affect other methods which calculate other statistics (e.g. the shortest book read ever)